include/swift/Remote/MetadataReader.h

```diff
@@ -2104,9 +2104,13 @@ class MetadataReader {
 
   Demangle::NodePointer
   buildContextDescriptorMangling(const ParentContextDescriptorRef &descriptor,
-                                 Demangler &dem) {
+                                 Demangler &dem, int recursion_limit) {
+    if (recursion_limit <= 0) {
+      return nullptr;
+    }
+
     if (descriptor.isResolved()) {
-      return buildContextDescriptorMangling(descriptor.getResolved(), dem);
+      return buildContextDescriptorMangling(descriptor.getResolved(), dem, recursion_limit - 1);
```

Comment:
You're cutting the actual recursion limit in half by decrementing here too, since every time we try to resolve a parent we'll end up decrementing recursion_limit twice.

Reply:
Good catch.  Fixed.